source/simulator/newton.cc

```diff
@@ -161,6 +161,239 @@ namespace aspect
   {
     newton_derivative_scaling_factor = set_newton_derivative_scaling_factor;
   }
+
+  template <int dim>
+  PreconditionerStabilization
+  NewtonHandler<dim>::
+  get_preconditioner_stabilization() const
+  {
+    return preconditioner_stabilization;
+  }
+
+
+  template <int dim>
+  void
+  NewtonHandler<dim>::
+  set_preconditioner_stabilization(const PreconditionerStabilization preconditioner_stabilization_)
+  {
+    preconditioner_stabilization = preconditioner_stabilization_;
+  }
+
+  template <int dim>
+  VelocityBlockStabilization
+  NewtonHandler<dim>::
+  get_velocity_block_stabilization() const
+  {
+    return velocity_block_stabilization;
+  }
+
+
+  template <int dim>
+  void
+  NewtonHandler<dim>::
+  set_velocity_block_stabilization(const VelocityBlockStabilization velocity_block_stabilization_)
+  {
+    velocity_block_stabilization = velocity_block_stabilization_;
+  }
+
+  template <int dim>
+  bool
+  NewtonHandler<dim>::
+  get_use_Newton_failsafe()
+  {
+    return use_Newton_failsafe;
+  }
+
+  template <int dim>
+  double
+  NewtonHandler<dim>::
+  get_nonlinear_switch_tolerance()
+  {
+    return nonlinear_switch_tolerance;
+  }
+
+  template <int dim>
+  unsigned int
+  NewtonHandler<dim>::
+  get_max_pre_newton_nonlinear_iterations()
+  {
+    return max_pre_newton_nonlinear_iterations;
+  }
+
+  template <int dim>
+  unsigned int
+  NewtonHandler<dim>::
+  get_max_newton_line_search_iterations()
+  {
+    return max_newton_line_search_iterations;
+  }
+
+  template <int dim>
+  bool
+  NewtonHandler<dim>::
+  get_use_newton_residual_scaling_method()
+  {
+    return use_newton_residual_scaling_method;
+  }
+
+  template <int dim>
+  double
+  NewtonHandler<dim>::
+  get_maximum_linear_stokes_solver_tolerance()
+  {
+    return maximum_linear_stokes_solver_tolerance;
+  }
+
+  template <int dim>
+  std::string
+  NewtonHandler<dim>::
+  get_preconditioner_stabilization_string(PreconditionerStabilization preconditioner_stabilization)
+  {
+    switch (preconditioner_stabilization)
+      {
+        case PreconditionerStabilization::SPD:
+          return "SPD";
+          break;
+        case PreconditionerStabilization::PD:
+          return "PD";
+          break;
+        case PreconditionerStabilization::symmetric:
+          return "symmetric";
+          break;
+        case PreconditionerStabilization::none:
+          return "none";
+          break;
+      }
+  }
```

Comment:
You'll want to put a
```
  default:
    Assert (false, ExcNotImplemented());
    return "";
```
case into the `switch` to make sure you return *something* and that you catch potential mistakes.